x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/column_types.ts

```diff
@@ -5,11 +5,13 @@
  */
 
 import type { Operation } from '../../../types';
+import { TimeScaleUnit } from '../../time_scale';
 
 export interface BaseIndexPatternColumn extends Operation {
   // Private
   operationType: string;
   customLabel?: boolean;
+  timeScale?: TimeScaleUnit;
```

Comment:
Why did you add this to the top-level `IndexPatternColumn` type instead of to `params`? We have more uses of `column.params`, including when building the expression output.

Reply:
I thought it would be a better place to clarify it's not managed by the individual operations, but it makes more sense to put it there anyway. I will adjust the PR.

Reply:
@wylieconlon Looking into this I would like to keep it as it is as the types for `params` are pretty complex right now and I would have to overwrite them in weird ways in a bunch of places to make it work. We should clean that up but I don't think we have to do it on this PR. Ideally I would like to keep the frame managed pieces to the top level, with `params` being reserved for the individual operations. This makes the types much simpler.